lib/BackgroundJob/RetentionJob.php

```diff
@@ -85,7 +85,7 @@ public function run($argument): void {
 		// Validate if tag still exists
 		$tag = $argument['tag'];
 		try {
-			$this->tagManager->getTagsByIds($tag);
+			$this->tagManager->getTagsByIds((string) $tag);
```

Comment:
should this maybe also be done in the createNamedParameter call? or once line 86

Reply:
No, because the retention code is int based. Only the TagManager code is string based although it casts to int afterwards itself as the DB is int for oc_systemtags* tables as well...